roachtest: test run

RaduBerinde opened the pull request:
#### issues: make IssueFormatter an interface

Epic: none
Release note: None

#### issues: use a bullet list for parameters

Change the formatting of parameters to use a bullet list.

Epic: none
Release note: None

#### issues: add top-level notes

Add functionality to optionally include a top-level note in the issue
description.

Epic: none
Release note: None

#### roachtest: segregate issues for coverage and metamorphic builds

Issues posted for metamorphic/coverage builds now have
`B-metamorphic`/`B-coverage` labels and no `release-blocker` label.
We also include top-level notices in the issue description with
information about these builds.

We adopt an existing issue only when the metamorphic/coverage labels
match exactly.

Informs: #114615
Release note: None

#### roachtest: allow posting issues from non-release branch

RaduBerinde committed:
roachtest: segregate issues for coverage and metamorphic builds

Issues posted for metamorphic/coverage builds now have
`B-metamorphic`/`B-coverage` labels and no `release-blocker` label.
We also include top-level notices in the issue description with
information about these builds.

We adopt an existing issue only when the metamorphic/coverage labels
match exactly.

Informs: #114615
Release note: None

RaduBerinde committed:
roachtest: allow posting issues from non-release branch

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/115199)
<!-- Reviewable:end -->

RaduBerinde commented:
https://teamcity.cockroachdb.com/viewQueued.html?itemId=12880802

RaduBerinde committed:
issues: use a bullet list for parameters

Change the formatting of parameters to use a bullet list.

Epic: none
Release note: None

RaduBerinde committed:
issues: make IssueFormatter an interface

Epic: none
Release note: None

RaduBerinde committed:
roachtest: clean up TestCreatePostRequest

Clean up this test to use explicit field keys when initializing
structs, otherwise changing the fields is a nightmare.

Epic: none
Release note: None

RaduBerinde committed:
issues: add top-level notes

Add functionality to optionally include a top-level note in the issue
description.

Epic: none
Release note: None

RaduBerinde committed:
issues: fix and clean up label logic

This change cleans up the logic around the labels of issues. We now
have an explicit list of labels in `PostRequest`, which defaults to
`C-test-failure,release-blocker` if not set. This makes things easier
and fixes a current bug where roachtest issues have `release-blocker`
even when the test spec has `NoReleaseBlocker` set.

We also add functionality for requiring a set of labels to match
before adopting queries. This will be used in an upcoming roachtest
change to separate issues filed in coverage or metamorphic build
modes.

Epic: none
Release note: None
